src/farmer.rs

```diff
@@ -179,8 +186,8 @@ enum FarmerCommand {
 
 pub struct Farmer {
     cmd_sender: oneshot::Sender<FarmerCommand>,
-    plot_directories: Vec<PathBuf>,
     reward_address: PublicKey,
+    plot_info: HashMap<PathBuf, Plot>,
```

Comment:
we are storing both PathBuf and Plot in here, but Plot also stores PathBuf in itself. Seemed weird to me?

Reply:
I mean, we are storing the same info twice just for ergonomics. That makes me think maybe there is a better design? Not a blocker for this PR on my end, that's why approved, but I still have the bias towards there is room for improvement in the design, if we have to store the same thing twice for ergonomics